### PR TITLE
RavenDB-8677 Delete WaitForNonStaleResultsAsOfNow, update server to w…

### PR DESCRIPTION
**Pull request by IdanHaim**
…ait if WaitForNonStaleResults is true

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 1]
```diff
@@ -2163,9 +2163,16 @@ private QueryDoneRunning MarkQueryAsRunning(IIndexQuery query, OperationCancelTo
 
         private static bool WillResultBeAcceptable(bool isStale, IndexQueryBase<BlittableJsonReaderObject> query, AsyncWaitForIndexing wait)
         {
+           
             if (isStale == false)
                 return true;
 
+            if (query.WaitForNonStaleResults && query.WaitForNonStaleResultsTimeout == null)
+            {
+                query.WaitForNonStaleResultsTimeout = TimeSpan.FromSeconds(30);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Indexes/Index.cs:2172]: This should be configurable?
Also this removes the ability to wait indefinitely for the results (by default in C# we are using DefaultTimeout (15sec)).

File: src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs [hunk 2]
```diff
@@ -124,16 +124,5 @@ public interface IDocumentQueryCustomization
         /// </param>
         /// <param name="waitTimeout">Maximum time to wait for index query results to become non-stale before exception is thrown.</param>
         IDocumentQueryCustomization WaitForNonStaleResultsAsOf(long cutOffEtag, TimeSpan waitTimeout);
-
-        /// <summary>
-        ///     Instructs the query to wait for non stale results as of now.
-        /// </summary>
-        IDocumentQueryCustomization WaitForNonStaleResultsAsOfNow();
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs:131]: Description/Summary from this should be moved to WaitForNonStaleResults?

File: src/Raven.Client/Documents/Queries/IndexQuery.cs [hunk 3]
```diff
@@ -166,12 +165,7 @@ public int PageSize
         }
 
         /// <summary>
-        /// Used to calculate index staleness. When set to <c>true</c> CutOff will be set to DateTime.UtcNow on server side.
-        /// </summary>
-        public bool WaitForNonStaleResultsAsOfNow { get; set; }
-
-        /// <summary>
-        /// CAUTION. Used by IDocumentSession ONLY. It will have NO effect if used with IDatabaseCommands or IAsyncDatabaseCommands.
+        /// When set to <c>true</c>> server side will wait until result are non stale or until timeout
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Queries/IndexQuery.cs:168]: We should leave the comment/summary from WaitForNonStaleResultsAsOfNow here?

Comment (IdanHaim) [reply src/Raven.Client/Documents/Queries/IndexQuery.cs:168]: We don't, It's not true anymore

File: src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs (hunk 2 above)
Comment (IdanHaim) [reply src/Raven.Client/Documents/Session/IDocumentQueryCustomization.cs:131]: We can do it with little changes

File: src/Raven.Client/Documents/Queries/IndexQuery.cs (hunk 3 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Queries/IndexQuery.cs:168]: The behavior of WaitForNonStaleResults now matches the behavior from WaitForNonStaleResultsAsOfNow so the description from WaitForNonStaleResultsAsOfNow should be used?

Comment (IdanHaim) [reply src/Raven.Client/Documents/Queries/IndexQuery.cs:168]: It wasn't accurate before

File: src/Raven.Server/Documents/Indexes/Index.cs (hunk 1 above)
Comment (IdanHaim) [reply src/Raven.Server/Documents/Indexes/Index.cs:2172]: I think to wait 30 seconds It's more then enough and if someone want to wait indefinitely he always can initialize the timeout.
Maybe we can decrease it to DefaultTimeout (15sec)).

Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Index.cs:2172]: Should be reduced, yes. And the user can control this by explicitly specifying the timeout.

**Pull request merged**